src/models/map.rs

```diff
@@ -58,63 +61,56 @@ impl Map {
     pub fn connect<
         M: 'static + Messenger + Clone + Send,
         PP: 'static + PacketProcessor + Clone + Send,
+        PA: 'static + PatchworkState + Clone + Send,
```

Comment:
Establishing a new pattern here for async stuff- basically throw a copy of your own sender so they can call you back once it's done. There might be better ways of doing this, not really sure about the adoption of callbacks here